python-connectors/kayrros_retrieve-dataset/connector.py

```diff
@@ -0,0 +1,123 @@
+# This file is the actual code for the custom Python dataset kayrros_retrieve-dataset
+
+from dataiku.connector import Connector
+from utils.authentification import get_headers
+import requests
+import dataiku
+
+class MyConnector(Connector):
+    
+    # A custom Python dataset is a subclass of connector.
+
+    def __init__(self, config, plugin_config):
+        
+        """
+        The configuration parameters set up by the user in the settings tab of the
+        dataset are passed as a json object 'config' to the constructor.
+        The static configuration parameters set up by the developer in the optional
+        file settings.json at the root of the plugin directory are passed as a json
+        object 'plugin_config' to the constructor
+        """
+        
+        Connector.__init__(self, config, plugin_config) 
+
+        # Perform some more initialization
+        self.id = self.config["id_dataset"]
+        self.preset = self.config["preset"]
+        self.username = self.preset["username"]
+        self.password = self.preset["password"]        
+        
+        
+    def get_read_schema(self):
+
+        return {"columns" : [ {"name": "value_date", "type" : "string"}, 
+                              {"name" :"metrics", "type" : "array"},
+                              {"name" :"extra_props", "type" : "array"},
+                              {"name" :"name", "type" : "string"}]}
+
+
+    def generate_rows(self, dataset_schema=None, dataset_partitioning=None,
+                            partition_id=None, records_limit = -1):
+        
+        """
+        The main reading method.
+
+        Returns a generator over the rows of the dataset (or partition)
+        Each yielded row must be a dictionary, indexed by column name.
+
+        The dataset schema and partitioning are given for information purpose.
+        """
+        
+        # I. Get the dataset from the id
+        
+        url_asset = "https://platform.api.kayrros.com/v1/processingresult/dataset/" + self.id
+
+        req = requests.post(url_asset, headers = get_headers(self.config, self.plugin_config))
+
+        if req.status_code == 200:
+            content = req.json()
+        
+        else:
+            raise Exception("Error retrieving dataset content")
```

Comment:
Make the error more explicit.

```suggestion
        try:        
            response = requests.post(url_asset, headers = get_headers(self.config, self.plugin_config))
            response.raise_for_status()
        except requests.exceptions.RequestException as error:
            logger.exception("Dataset could not be retrieved because of the following error:\n {}".format(error))
            raise(error)
        content = response.json()
        logger.info("Received {} records".format(len(content["assets"])))
```